### PR TITLE
Add RELEASE_INFRA_PUSH related code to support pushes from kubernetes/release.

### PR DESCRIPTION
**Pull request by david-mcmahon**
ref #16529

### CODE REVIEW

File: hack/jenkins/build.sh [hunk 1]
```diff
@@ -54,9 +57,25 @@ git clean -fdx
...
 go run ./hack/e2e.go -v --build
 
-[[ ${KUBE_SKIP_PUSH_GCS:-} =~ ^[yY]$ ]] || {
-    # Push to GCS
-    ./build/push-ci-build.sh
-}
+# Push to GCS?
+if [[ ${KUBE_SKIP_PUSH_GCS:-} =~ ^[yY]$ ]]; then
+  echo "Not pushed to GCS..."
+elif $RELEASE_INFRA_PUSH; then
```
Comment (spxtr) [inline hack/jenkins/build.sh]: `[[ -n "${RELEASE_INFRA_PUSH:-}" ]]`

File: hack/jenkins/build.sh [hunk 2]
```diff
@@ -54,9 +57,25 @@ git clean -fdx
...
+  git clone https://github.com/kubernetes/release $WORKSPACE/release
+  RELEASE_INFRA_CLONE=$WORKSPACE/release
```
Comment (spxtr) [inline hack/jenkins/build.sh]: `readonly release_infra_clone="${WORKSPACE}/release"`

File: hack/jenkins/build.sh [hunk 3]
```diff
@@ -54,9 +57,25 @@ git clean -fdx
...
+
+  if [[ ! -x $RELEASE_INFRA_CLONE/push-ci-build.sh ]]; then
+    echo "FATAL: Something went wrong." \
```
Comment (spxtr) [inline hack/jenkins/build.sh]: `>&2`

File: hack/jenkins/build.sh [hunk 4]
```diff
@@ -54,9 +57,25 @@ git clean -fdx
...
+         "$RELEASE_INFRA_CLONE/push-ci-build.sh isn't available. Exiting..."
+    exit 1
+  fi
+
+  $FEDERATION && federation_flag="--federation"
+  # Use --nomock to do the real thing
+  #$RELEASE_INFRA_CLONE/push-ci-build.sh --nomock $federation_flag
+  $RELEASE_INFRA_CLONE/push-ci-build.sh $federation_flag
```
Comment (spxtr) [inline hack/jenkins/build.sh]: Will this break with `nounset`?

File: hack/jenkins/build.sh [hunk 5]
```diff
@@ -54,9 +57,25 @@ git clean -fdx
...
+  git clone https://github.com/kubernetes/release $WORKSPACE/release
+  RELEASE_INFRA_CLONE=$WORKSPACE/release
+
+  if [[ ! -x $RELEASE_INFRA_CLONE/push-ci-build.sh ]]; then
+    echo "FATAL: Something went wrong." \
+         "$RELEASE_INFRA_CLONE/push-ci-build.sh isn't available. Exiting..."
+    exit 1
+  fi
+
+  $FEDERATION && federation_flag="--federation"
```
Comment (spxtr) [inline hack/jenkins/build.sh]: This will probably break with `nounset`.

**Comment by spxtr** [comment]
For testing this, you can edit the file to set whatever environment variables will make it do its thing and then be sure to remove them once you've seen it work.

File: hack/jenkins/build.sh (hunk 1 above)
Comment (david-mcmahon) [inline hack/jenkins/build.sh]: Well RELEASE_INFRA_PUSH is a convenient true or false bool so we can code it so it reads nice like that.
The other (unsightly) option is:
if [[ "${RELEASE_INFRA_PUSH:-}" == "true" ]]
But -n will always be true unless we change the default.

Comment (david-mcmahon) [inline hack/jenkins/build.sh]: Another nice option in bash is to use 0 or 1 for your bool values and take advantage of the double paren eval.  Both are certainly more Go or Python like than the overly decorated expression above (== true):

So both read nicely:
```
RELEASE_INFRA_PUSH=1
if ((RELEASE_INFRA_PUSH)); then
```
Or:
```
RELEASE_INFRA_PUSH=true
if $RELEASE_INFRA_PUSH; then
```

**Comment by david-mcmahon** [comment]
ping.

**Comment by spxtr** [comment]
cc @rmmh test-infra on-call

File: hack/jenkins/build.sh (hunk 1 above)
Comment (spxtr) [inline hack/jenkins/build.sh]: Okay, seems reasonable. It's just different from the norm in this repo.

File: hack/jenkins/build.sh [hunk 6]
```diff
@@ -54,9 +57,25 @@ git clean -fdx
...
+  if [[ ! -x $release_infra_clone/push-ci-build.sh ]]; then
+    echo "FATAL: Something went wrong." \
+         "$release_infra_clone/push-ci-build.sh isn't available. Exiting..." >&2
+    exit 1
+  fi
+
+  $FEDERATION && federation_flag="--federation"
+  # Use --nomock to do the real thing
+  #$release_infra_clone/push-ci-build.sh --nomock $federation_flag
+  $release_infra_clone/push-ci-build.sh $federation_flag
```
Comment (spxtr) [inline hack/jenkins/build.sh]: If `$FEDERATION` is false then `federation_flag` never gets set and this will fail from nounset, I think.

Comment (david-mcmahon) [inline hack/jenkins/build.sh]: PTAL

**Comment by spxtr** [comment]
Ok, I think it's reasonable.

File: hack/jenkins/build.sh (hunk 1 above)
Comment (ixdy) [inline hack/jenkins/build.sh]: minor nit - scripts in this repo generally use `${VAR}` rather than `$VAR`.

Comment (david-mcmahon) [inline hack/jenkins/build.sh]: Indeed.  Will correct next iteration.

**Pull request merged**